### PR TITLE
upgrading to go version 1.17 

### PR DESCRIPTION
**Pull request by vipragupta**
- upgrading to go version 1.17 
- updating the monitor tags map from interface value to string value

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -83,13 +84,13 @@ func (status HealthStatus) String() string {
 	return "unknown"
 }
 
-func GetInstanceStatusTag(svcContainer SvcContainer, healthy HealthStatus, running bool, exitCode int) map[string]interface{} {
-	return map[string]interface{}{
+func GetInstanceStatusTag(svcContainer SvcContainer, healthy HealthStatus, running bool, exitCode int) map[string]string {
+	return map[string]string{
```
Comment (ridv) [inline types/types.go:88]: What's the rational behind this change? Will it cause issues upstream for us?

File: go.mod [hunk 2]
```diff
@@ -33,9 +33,22 @@ require (
...
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk => github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk v0.14.0
-	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver => github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver v0.14.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig => github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig v0.46.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk => github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk v0.46.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver => github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver v0.46.0
+	go.opentelemetry.io/collector => go.opentelemetry.io/collector v0.46.0
+    github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter => github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter v0.46.0
```
Comment (JiaminZhu) [inline go.mod]: nit: please fix the indentation.

**Review by JiaminZhu — COMMENTED** [review]
LGTM for the changes I requsted.

File: types/types.go (hunk 1 above)
Comment (vipragupta) [reply types/types.go:88]: this is with regards to otel version upgrade. otel tags no longer accept a map of string-interface

**Review by ridv — APPROVED** [review]
:shipit:

**Review by JiaminZhu — APPROVED** [review]

**Pull request merged**